service/src/main/java/bio/terra/cbas/controllers/RunsApiController.java

```diff
@@ -63,4 +75,35 @@ public ResponseEntity<RunLogResponse> getRuns(UUID runSetId) {
         new RunLogResponse().runs(responseList).fullyUpdated(updatedRunsResult.fullyUpdated()),
         HttpStatus.OK);
   }
+
+  @Override
+  public ResponseEntity<Void> postRunResults(RunResultsRequest body) {
+    // validate user permission
+    // check if current user has read permissions on the workspace
+    if (!samService.hasWritePermission()) {
+      throw new ForbiddenException(SamService.WRITE_ACTION, SamService.RESOURCE_TYPE_WORKSPACE);
+    }
+
+    // validate request
+    UUID runId = body.getWorkflowId();
+    CbasRunStatus resultsStatus = CbasRunStatus.fromValue(body.getState());
```

Comment:
I think you'll need `CbasRunStatus.fromCromwellStatus` here.

Reply:
Addressed